src/Akeneo/Pim/Enrichment/Product/back/Application/Applier/AssociationUserIntentCollectionApplier.php

```diff
@@ -59,8 +61,9 @@ public function apply(UserIntent $userIntent, ProductInterface $product, int $us
                     continue;
                 }
 
+                $nonViewableProducts = $this->getNonViewableProducts->fromProductIdentifiers($formerAssociations, $userId);
                 $normalizedAssociations[$associationUserIntent->associationType()][$entityType] = \array_values(
-                    \array_unique($associationUserIntent->productIdentifiers())
+                    \array_merge($nonViewableProducts, \array_unique($associationUserIntent->productIdentifiers()))
```

Comment:
```suggestion
                    \array_unique(\array_merge($nonViewableProducts, $associationUserIntent->productIdentifiers()))
```

it should be the same, but it's safer like this IMO